routes/auth.routes.js

```diff
@@ -0,0 +1,94 @@
+// routes/auth.routes.js
+const { Router } = require("express");
+const router = new Router();
+const bcryptjs = require("bcryptjs");
+const saltRounds = 11;
+const User = require("../models/User.model");
+const mongoose = require("mongoose");
+// .get() route ==> to display the signup form to users
+router.get("/signup", (req, res) => res.render("auth/signup"));
+// .post() route ==> to process form data
+router.post("/signup", (req, res, next) => {
+  const { username, email, password } = req.body;
+
+  if (!username || !email || !password) {
+    res.render("auth/signup", {
+      errorMessage:
+        "All fields are mandatory. Please provide your username, email and password.",
+    });
+    return;
+  }
+  const regex = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}/;
+  if (!regex.test(password)) {
+    res.status(500).render("auth/signup", {
+      errorMessage:
+        "Password needs to have at least 6 chars and must contain at least one number, one lowercase and one uppercase letter.",
+    });
+    return;
+  }
+  bcryptjs
+    .genSalt(saltRounds)
+    .then((salt) => bcryptjs.hash(password, salt))
+    .then((hashedPassword) => {
+      return User.create({
+        username,
+        email,
+        passwordHash: hashedPassword,
+      });
+    })
+    .then((userFromDB) => {
+      console.log("Newly created user is: ", userFromDB);
```

Comment:
If you redirect users to their profile then you should start a session here so they don't need to login.
```
.then((userFromDB) => {
req.session.userInSession = userFromDB;
res.redirect("/userProfile");
    })